src/generic/AbstractBandedMatrix.jl

```diff
@@ -336,3 +336,66 @@ if VERSION >= v"1.9"
     copy(A::Adjoint{T,<:AbstractBandedMatrix}) where T = copy(parent(A))'
     copy(A::Transpose{T,<:AbstractBandedMatrix}) where T = transpose(copy(parent(A)))
 end
+
+function sum!(ret::AbstractArray, A::AbstractBandedMatrix)
+    #Behaves similarly to Base.sum!
+    ret .= 0
+    n,m = size(A)
+    s = size(ret)
+    l = length(s)
+    #Check for singleton dimension and perform respective sum
+    if s[1] == 1 && (l == 1 || s[2]==1)
+        for j = 1:m, i = colrange(A, j)
+            ret .+= A[i, j]
+        end
+    elseif s[1] == n && (l == 1 || s[2]==1)
+        for i = 1:n, j = rowrange(A, i)
+            ret[i, 1] += A[i, j]
+        end
+    elseif s[1] == 1 && s[2] == m
+        for j = 1:m, i = colrange(A, j)
+            ret[1, j] += A[i, j]
+        end
+    elseif s[1] == n && s[2] == m
+        ret = A
```

Comment:
This is not changing `ret`!
```suggestion
        copyto!(ret, A)
```